CHANGES.md

```diff
@@ -1,3 +1,20 @@
+# grmtools 0.10.3 (XXXX-YY-ZZ)
+
+* `CTBuilder::process_file_in_src` is deprecated in favour of
+  `CTBuilder::grammar_path_in_src` and `process`. In most cases you can change
```

Comment:
I'm unsure about the name `grammar_path_in_src`. To me it suggest we give it a source which contains the grammar path. Maybe it should just be called `grammar_src`, since we are passing a grammar file, not a path.

Reply:
I'm also not sure what the best name is, and I was curious what you would think! I think `grammar_src` isn't good because that can be confused with `grammar_path`: I think we need to get the notion of "this file is relative to `src/`" across somehow (but without being too wordy). Maybe something like `grammar_file_in_src` or even `find_grammar_in_src`?

Reply:
After reading the rest of the commit I realised that `src` was referencing the project's `src` directory. However, that is not always correct. What is actually used is the current working directory which isn't necessarily the project root. Maybe `grammar_file_in_cwd` would me more accurate, though it is a bit of a mouthful.

What does `grammar_path` take as an argument? I think you called it `path` because it takes a rust `Path`, but is that clear enough to the end user? When I read `path` I assume it takes a path and not for example a file.

Reply:
> After reading the rest of the commit I realised that src was referencing the project's src directory. However, that is not always correct. What is actually used is the current working directory which isn't necessarily the project root.

Are there cases when the CWD isn't the project root? If so, is there a way to genuinely use the project root?

> What does grammar_path take as an argument?

`AsRef<Path>`. It's a common idiom in the Rust libraries, but a bit hard to describe succinctly, because it can auto-transform strings to `Path`s.

Reply:
> `AsRef<Path>`. It's a common idiom in the Rust libraries, but a bit hard to describe succinctly, because it can auto-transform strings to `Path`s.

I meant, what does the user pass in, not the type, but the destination. Do they pass a file or really just a path. I think it's the former, that's why I pointed out that it might be confusing to the end user, that the function is called `grammar_path` when actually they are passing in a file.

Reply:
Technically they pass in a relative path.

Reply:
To a file or directory?

Reply:
Can I suggest we address the questions to the methods' doc strings? `CHANGES.md` only includes a summary, not the complete specification.

Reply:
What questions? I didn't see any in the doc strings. Anyway, I was just trying to help find clearer names for those methods. If you don't want to change them, that's fine.

Reply:
I mean: if you find the doc strings for these functions (they're all in this PR, as they've been changed), you might find that it helps give you ideas for the names. e.g. https://github.com/softdevteam/grmtools/blob/9cae583549012f11fe534a6f5f0ba608279ef535/lrpar/src/lib/ctbuilder.rs#L200

Reply:
The only ideas I got left is maybe `local_grammar_file`, `relative_grammar_file`. My mind is a bit blank at the moment.

Reply:
I think `grammar_in_src_dir` might be the best bet?

Reply:
I can live with that.

Reply:
See if https://github.com/softdevteam/grmtools/pull/247/commits/da8d4490ba6869892d189e4bb0b29ed7bec9a0bd works for you.